### ISSUE TITLE
Crash on Windows 10 when connecting to an IRC server

### ISSUE THREAD

**Issue by peterholak** [hexchat/hexchat#1475]
I'm using HexChat 2.10.2 x64. The same version that worked fine on Windows 8.1 now crashes on Windows 10 with the following message:

`GLib-WARNING **: Need to have NULL encoding to set the buffering state of the channel.`

I tried building it from source and running it with gdb, it seems to be crashing on line 221 in `src/common/util.c` (`g_io_channel_set_buffered (source, FALSE);`)

**Comment by Arnavion** [comment]
Seems they should be swapped.

**Comment by peterholak** [comment]
Swapping them results in crash with `GLib-WARNING **: Need to set the channel buffered before setting the encoding.`

**Comment by peterholak** [comment]
Putting the `g_io_channel_set_encoding` first and putting both lines inside `if (g_io_channel_get_buffered(source) != FALSE || g_io_channel_get_encoding(source) != NULL) {` gets rid of the crash (and both lines are called exactly once).

**Comment by peterholak** [comment]
After looking into it a bit more, it seems for some reason it's behaving on my computer as if it were run with the `--g-fatal-warnings` argument (which it is not).

I'll check if it's some environment variable or config file somewhere, but I'm 100% sure I didn't create one manually, so it might be the work of some other program on my computer. Either way, it means this doesn't affect a clean Win10 install.

**Comment by Arnavion** [comment]
No, a warning is still indicative of issues, so it should still be fixed.
